hibernate-core/src/main/java/org/hibernate/dialect/FirebirdDialect.java

```diff
@@ -114,23 +117,39 @@ public FirebirdDialect(int version) {
 		//no precision for 'timestamp' type
 		registerColumnType( Types.TIMESTAMP, "timestamp" );
 		if ( getVersion() < 400 ) {
+			// No time zone support, map to without time zone types
 			registerColumnType( Types.TIMESTAMP_WITH_TIMEZONE, "timestamp" );
+			registerColumnType( Types.TIME_WITH_TIMEZONE, "time" );
 		}
 		else {
 			registerColumnType( Types.TIMESTAMP_WITH_TIMEZONE, "timestamp with time zone" );
 		}
+		// Single byte character sets can be 32_765 characters, but assume use of UTF8
 		registerColumnType( Types.VARCHAR, 8_191, "varchar($l)" );
 		registerColumnType( Types.VARCHAR, "blob sub_type text" );
 
-		registerColumnType( Types.BINARY, 32_767, "char($l) character set octets" );
+		if ( getVersion() < 400 ) {
+			registerColumnType( Types.BINARY, 32_767, "char($l) character set octets" );
+		}
+		else {
+			registerColumnType( Types.BINARY, 32_767, "binary($l)" );
+		}
 		registerColumnType( Types.BINARY, "blob sub_type binary" );
 
-		registerColumnType( Types.VARBINARY, 32_765, "varchar($l) character set octets" );
+		if ( getVersion() < 400 ) {
+			registerColumnType( Types.VARBINARY, 32_765, "varchar($l) character set octets" );
+		}
+		else {
+			registerColumnType( Types.VARBINARY, 32_765, "varbinary($l)" );
+		}
 		registerColumnType( Types.VARBINARY, "blob sub_type binary" );
 
 		registerColumnType( Types.BLOB, "blob sub_type binary" );
 		registerColumnType( Types.CLOB, "blob sub_type text" );
 		registerColumnType( Types.NCLOB, "blob sub_type text" ); // Firebird doesn't have NCLOB, but Jaybird emulates NCLOB support
+		registerColumnType( Types.LONGVARBINARY, "blob sub_type binary" );
+		registerColumnType( Types.LONGVARCHAR, "blob sub_type text" );
+		registerColumnType( Types.LONGNVARCHAR, "blob sub_type text" );
```

Comment:
I would prefer not to add these. We don't use them anymore, and except for a couple of exceptions the other `Dialect`s generally don't register any mapping for them.

Reply:
I wanted to be complete, as Jaybird identifies `blob sub_type binary` and `blob sub_type text` as Types.LONGVARBINARY and Types.LONGVARCHAR in its metadata methods. I haven't found anything where this mattered, but I was wondering if there was a case where some form of metadata inspection would need this info.

If you're not aware of such a case either, I'll remove it again.

Reply:
These column type registrations are only used for DDL generation. We don't have any types that use these.

It is in principle possible for a `UserType` to return `Types.LONGVARBINARY` but it wouldn't make much sense here: it would be more sensible to return `Types.BLOB` if they want a `blob`.

Reply:
Ok, I have removed them